lua/framework/logging/log.cc

```diff
@@ -1,7 +1,7 @@
 // SPDX-FileCopyrightText: 2024 The OpenSn Authors <https://open-sn.github.io/opensn/>
 // SPDX-License-Identifier: MIT
 
-#include "lua_log.h"
+#include "log.h"
 #include "framework/logging/log.h"
```

Comment:
could it be confusing to have 2 files named log.h in different folders? wouldn't it be better to have different filenames for clarity??

Reply:
This should actually use the full path "lua/framework/logging/log.h", that would avoid the potential confusion.
Our search path for header files is set to the repo root, so compiler will always pick up the right file here, i.e. the one from the active directory.
And yes, situations like this were probably the driver for the lua prefix/postfix.

Reply:
@zhardy-lanl @wdhawkins @andrsd : so what do you think about filenames (.cc or .h) being the same in the lua/ and in the rest of the src code?

Reply:
I agree with @andrsd that the full path should be used to avoid confusion and am in full support of removing `_lua` or `lua_` from file names. Things in the Lua folder should be assumed to be related to Lua.

Reply:
Ok. So is this PR making sure that the full path is used or should there be a new issue?

Reply:
This PR is about removing the lua prefix from file names. You just happen to spot another problem that needs to be fixed...

Reply:
created isse here: #283 
will resolve this conversation and approve this PR now